src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,94 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int CAPACITY_MULTIPLIER = 2;
+    private Node<K, V>[] nodesArray;
+    private int capacity;
+    private int size;
+
+    public MyHashMap() {
+        nodesArray = new Node[DEFAULT_CAPACITY];
+        capacity = DEFAULT_CAPACITY;
+    }
+
+    private int getHash(Object key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode() % capacity);
+    }
+
+    private void resize() {
+        size = 0;
+        Node<K, V>[] tempNodesArray = nodesArray;
+        capacity *= CAPACITY_MULTIPLIER;
+        nodesArray = new Node[capacity];
+        for (Node<K, V> node: tempNodesArray) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == capacity * LOAD_FACTOR) {
+            resize();
+        }
+        int index = getHash(key);
+        Node<K, V> nodeToPut = new Node<>(index, key, value, null);
+        Node<K, V> currentNode = nodesArray[index];
+        if (currentNode == null) {
```

Comment:
The if statement checking if currentNode is null can be handled in the while loop condition